src/catalog/src/durable/impls/persist.rs

```diff
@@ -109,6 +113,14 @@ impl PersistHandle {
         const SEED: usize = 1;
         let shard_id = Self::shard_id(organization_id, SEED);
         debug!(?shard_id, "new persist backed catalog state");
+        let since_handle = persist_client
+            .open_critical_since(
+                shard_id,
+                PersistClient::CONTROLLER_CRITICAL_SINCE,
```

Comment:
No preference here - you've got a better sense of how this may evolve than I do. (Yesterday I was trying to puzzle out if every cluster would eventually want its own critical handle for the catalog so they don't have to coordinate between them...)

Reply:
Let's leave it as a TODO, there's precedent for that in https://github.com/MaterializeInc/materialize/blob/979b484bc855af3e29801f7229a8d555fbb1ad66/src/persist-txn/src/txns.rs#L159-L161

> Yesterday I was trying to puzzle out if every cluster would eventually want its own critical handle for the catalog so they don't have to coordinate between them...

I think Pv2 is also still trying to puzzle this out. In one world, the controllers would become a sort of "rich client" that lives locally in the cluster, which is nice for isolation. In that world, yeah exactly each cluster gets a critical since. But last I heard, I think we were leaning toward keeping the controllers as env-singletons and communicating with them over rpc (mostly for read holds)